src/nat.rs

```diff
@@ -32,8 +34,25 @@ pub fn get_public_ip_addr() -> Result<IpAddr, String> {
     }
 }
 
+pub fn udp_random_bind(start: u16, end: u16, mut tries: u32) -> Result<UdpSocket, String> {
```

Comment:
I'd prefer a loop and a non-mutable "tries", but maybe that's because I'm a C brogrammer

Reply:
definitely ditch the `mut`.  The existing structure could be kept with simply:
```rust
udp_random_bind(start, end, tries - 1)
```